app/javascript/packs/archive_modal.jsx

```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from "reactstrap";
+
+export class ConfirmationModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modal: false
+    };
+
+    this.toggle = this.toggle.bind(this);
+  }
+
+  toggle() {
```

Comment:
```suggestion
  toggle = () => {
```
you can use arrow function to skip binding function in constructor